lib/semantics/type.h

```diff
@@ -202,23 +225,47 @@ class GenericSpec {
   friend std::ostream &operator<<(std::ostream &, const GenericSpec &);
 };
 
-// The value of a len type parameter
-using LenParamValue = Bound;
+// A type parameter value: integer expression or assumed or deferred.
+class ParamValue {
+public:
+  static const ParamValue Assumed() { return ParamValue(Category::Assumed); }
+  static const ParamValue Deferred() { return ParamValue(Category::Deferred); }
```

Comment:
Should this be ParamValue { } ?

Reply:
Or just `{Category::...}`, which would be okay here from a readability standpoint since the type of the function result is right there on the same line.

Reply:
OK, I'll change this.